daemon/links/links.go

```diff
@@ -26,6 +27,11 @@ type Link struct {
 // EnvVars generates environment variables for the linked container
 // for the Link with the given options.
 func EnvVars(parentIP, childIP, name string, env []string, exposedPorts map[container.PortRangeProto]struct{}) []string {
+	// Environment variables defined when using legacy links are deprecated and will be removed in a future release.
```

Comment:
Perhaps as alternative, likely would fix some of the test failures, could we put the env-var on the caller side? I.e., don't call this function if the env-var isn't set

Reply:
Done. I also removed the broken integration-cli test that was checking legacy links, and skipped the broken docker-py test.